### PR TITLE
Milestone #7: Add forms

### PR DESCRIPTION
**Pull request by carlylechia**
#### The main goal of this milestone was to implement functionalities for adding comments and likes to posts. Below are the steps I used to do this:

- [x] Created a method called `current_user` in the **ApplicationController** which made current user data available to all controllers.

> It returns the first user from the database.

> NB: Proper authentication was not implemented.

- [x] Created forms to perform the following functions:

> Create a Post on behalf of the current_user (using the method that was created in the ApplicationController).

> Create a comment on behalf of the current_user (using the method that was created in the ApplicationController).

- [x] Users can now add likes and comments to Posts.

Thanks for your kind review 😄 !

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,4 +1,19 @@
...
+    respond_to do |format|
+      if @comment.save
+        format.html { redirect_to user_post_path(current_user, @post) }
+        flash[:success] = 'Comment created successfully'
+      else
+        flash[:error] = 'Failed to save comment'
+        format.html { redirect_to user_post_path(current_user, @post) }
+      end
+    end
+  end
```
Comment (mikethreels) [inline app/controllers/comments_controller.rb:22]: - [x] in the commentsController there should be a `new` action as well.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Carlyle, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (carlylechia) [reply app/controllers/comments_controller.rb:22]: Oh thanks @mikethreels, I forgot to add that. Doing that right away

**Review by elmar8287 — APPROVED** [review]
Hi, @carlylechia.

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 

### Highlights

- All suggestions are implemented 👍 

### Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
### Please, remember to tag me (@elmar8287) in your question so I can receive the notification 👍 
giphy

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by carlylechia** [comment]
Thanks for your kind review @elmar8287

**Pull request merged**